part_1/solution.py

```diff
@@ -0,0 +1,1327 @@
+</ul>
+Click on the link to view the tensorboard and it should open in your browser.
+</div>
+"""
+
+# %% Imports and paths tags=[]
+
+
+# Function to find an available port
+def find_free_port():
+    import socket
+
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.bind(("", 0))
+        return s.getsockname()[1]
+
+
+# Launch TensorBoard on the browser
+def launch_tensorboard(log_dir):
+    import subprocess
+
+    port = find_free_port()
+    tensorboard_cmd = f"tensorboard --logdir={log_dir} --port={port}"
+    process = subprocess.Popen(tensorboard_cmd, shell=True)
+    print(
+        f"TensorBoard started at http://localhost:{port}. \n"
+        "If you are using VSCode remote session, forward the port using the PORTS tab next to TERMINAL."
+    )
+    return process
+
+
+# Launch tensorboard and click on the link to view the logs.
+tensorboard_process = launch_tensorboard(log_dir)
+
+# %% [markdown]
+"""
+## Load OME-Zarr Dataset
+
+There should be 34 FOVs in the dataset.
+
+Each FOV consists of 3 channels of 2048x2048 images,
+saved in the [High-Content Screening (HCS) layout](https://ngff.openmicroscopy.org/latest/#hcs-layout)
+specified by the Open Microscopy Environment Next Generation File Format
+(OME-NGFF).
+
+- The layout on the disk is: `row/col/field/pyramid_level/timepoint/channel/z/y/x.`
+- These datasets only have 1 level in the pyramid (highest resolution) which is '0'.
+"""
+
+# %% [markdown]
+"""
+<div class="alert alert-warning">
+You can inspect the tree structure by using your terminal:
+<code> iohub info -v "path-to-ome-zarr" </code>
+
+<br>
+More info on the CLI:
+<code>iohub info --help </code> to see the help menu.
+</div>
+"""
+# %%
+# This is the python function called by `iohub info` CLI command
+print_info(data_path, verbose=True)
+
+# Open and inspect the dataset.
+dataset = open_ome_zarr(data_path)
+
+# %%
+# Use the field and pyramid_level below to visualize data.
+row = 0
+col = 0
+field = 9  # TODO: Change this to explore data.
+
+
+# NOTE: this dataset only has one level
+pyaramid_level = 0
+
+# `channel_names` is the metadata that is stored with data according to the OME-NGFF spec.
+n_channels = len(dataset.channel_names)
+
+image = dataset[f"{row}/{col}/{field}/{pyaramid_level}"].numpy()
+print(f"data shape: {image.shape}, FOV: {field}, pyramid level: {pyaramid_level}")
+
+figure, axes = plt.subplots(1, n_channels, figsize=(9, 3))
+
+for i in range(n_channels):
+    for i in range(n_channels):
+        channel_image = image[0, i, 0]
+        # Adjust contrast to 0.5th and 99.5th percentile of pixel values.
+        p_low, p_high = np.percentile(channel_image, (0.5, 99.5))
+        channel_image = np.clip(channel_image, p_low, p_high)
+        axes[i].imshow(channel_image, cmap="gray")
+        axes[i].axis("off")
+        axes[i].set_title(dataset.channel_names[i])
+plt.tight_layout()
+
+# %% [markdown]
+# <div class="alert alert-info">
+#
+# ### Task 1.1
```

Comment:
Generally I think it's easier to read if we put the task cells above the TODO, rather than below